### PR TITLE
1503 prometheus example

### LINKED ISSUE #1503: Create an example of using the Prometheus exporter

### PR DESCRIPTION
**Pull request by jarebudev**
resolves #1503 

Screenshot from 2020-08-17 22-01-16

### CODE REVIEW

File: examples/prometheus/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Prometheus Example
+
+This example demonstrates how to use the OpenTelemetry SDK 
+to instrument a simple application using Prometheus as the metric exporter and expose the metrics via HTTP. 
+
+These are collected by a Prometheus instance which is configured to pull these  metrics via HTTP.
```
Comment (thisthat) [inline examples/prometheus/README.md:6]:
```suggestion
These are collected by a Prometheus instance which is configured to pull these metrics via HTTP. 
```

File: examples/prometheus/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* Docker 19.03
+
+## 1 - Compile 
+```shell script
+../gradlew fatJar
+```
+## 2 - Run Prometheus
+
+Start Prometheus instance with a configuration that sets up a HTTP collection job for  ```127.0.0.1:19090```
+`
```
Comment (thisthat) [inline examples/prometheus/README.md:21]:
```suggestion

```

File: examples/prometheus/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+See [prometheus.yml](prometheus.yml)
+
+```shell script
+docker run --network="host" --rm -it \
+    --name prometheus \
+    -v $(pwd)/prometheus.yml:/etc/prometheus/prometheus.yml \
+    prom/prometheus
```
Comment (thisthat) [inline examples/prometheus/README.md:29]:
```suggestion
    -p 9090:9090 \
    prom/prometheus 
```

File: examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        incomingMessageCount = ThreadLocalRandom.current().nextLong(100);
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+      }
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+
+    int port = Integer.parseInt(args[0]);
```
Comment (thisthat) [inline examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java:71]: I would add a simple check that verifies the port is passed as an argument and prints a message if it is missing.

File: examples/prometheus/README.md (hunk 3 above)
Comment (jarebudev) [reply examples/prometheus/README.md:28]: I left the published ports out of the run command as these are ignored, as the container is run using ```--network="host"``` mode.

Comment (thisthat) [reply examples/prometheus/README.md:28]: Oh, I miss that! You are right! :D

File: examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java (hunk 4 above)
Comment (jarebudev) [reply examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java]: Done!

Comment (thisthat) [reply examples/prometheus/src/main/java/io/opentelemetry/example/PrometheusExample.java]: Thank you :)

**Comment by jkwatson** [comment]
Thanks for this contribution! Can you attach a screenshot of the prometheus UI from this example to the PR description?

**Comment by jarebudev** [comment]
Screenshot from 2020-08-17 22-01-16

**Comment by jkwatson** [comment]
Sorry.. I meant to put the image in the *description* of the PR, so people looking at it in the future will see it front-and center in the PR itself, without having to look at comments.

**Comment by jarebudev** [comment]
> Sorry.. I meant to put the image in the _description_ of the PR, so people looking at it in the future will see it front-and center in the PR itself, without having to look at comments.

haha i'll correct that! it's been a long day!

**Pull request merged**